# EDA on Project Development Cycle

* Over-sampling of released projects?
  * released percentage in ghtorrent data v.s. released in current data
* Number of commits by project age?

In [5]:
import dask.dataframe as dd
from dask import compute as dcompute
import os
data_dir = '/home/faithfeng/Dropbox/GitHub/Data/new_python_dataset'

In [2]:
pyprojects = dd.read_csv(data_dir+os.sep+'merged_pyprojects.csv')
commits = dd.read_csv(data_dir+os.sep+'pyauthors_full_commits.csv')
print(len(pyprojects))
print(len(commits))

369343
27262389


In [3]:
pyprojects.head()

project_id                                        description  \
0          67                       Chinese segmentation library   
1         119  Easy to use CLI for picture classification bas...   
2         197        Use the Twilio API to send hourly catfacts!   
3         220         GoodData client library written in python.   
4         265  A django rest framework for handling requests,...   

                                                 url  released  from_pypi  
0        https://api.github.com/repos/victorlin/loso       0.0        0.0  
1      https://api.github.com/repos/fudgefr/pyctozor       0.0        0.0  
2    https://api.github.com/repos/rossdylan/catfacts       0.0        0.0  
3  https://api.github.com/repos/comoga/gooddata-p...       0.0        0.0  
4         https://api.github.com/repos/cakey/Shimmer       0.0        0.0

In [6]:
# percentage of release from GHTorrent
ghrelease = pyprojects[pyprojects['from_pypi']==0]['released'].mean()
allrelease = pyprojects['released'].mean()
print(dcompute(ghrelease, allrelease))

(0.0703590349091926, 0.18993726698488939)


__It is clear that we have an over-sampling of released projects.__
* If we don't use PyPI data, the release percentage is about 7\%.
* After merging with PyPI data, the release percentage becomes 18.99\%.

In [4]:
commits.head()

author_id  project_id  commit_year  commit_month  num_commits
0          3         1.0         2011            12            2
1          3     45316.0         2010            12            2
2          3     45316.0         2011            11            1
3          3    199049.0         2012             1            4
4          3    330967.0         2012             4            1

In [ ]:
pycommits = commits.merge(how='inner', left_on=)